pgmpy/models/MarkovChain.py

```diff
@@ -0,0 +1,299 @@
+#!/usr/bin/env python3
+import numpy as np
+from pandas import DataFrame
+
+from pgmpy.utils import sample_discrete
+
+
+class MarkovChain(object):
+    """
+    Class to represent a Markov Chain with multiple kernels for factored state space,
+    along with methods to simulate a run.
+
+    Public Methods:
+    ---------------
+    set_start_state(state)
+    add_variable(variable, cardinality)
+    add_variables_from(vars_list, cards_list)
+    add_transition_model(variable, transition_dict)
+    sample(start_state, size)
+
+    Examples:
+    ---------
+    Create an empty Markov Chain:
+    >>> from pgmpy.models import MarkovChain as MC
+    >>> model = MC()
+
+    And then add variables to it
+    >>> model.add_variables_from(['intel', 'diff'], [2, 3])
+
+    Or directly create a Markov Chain from a list of variables and their cardinalities
+    >>> model = MC(['intel', 'diff'], [2, 3])
+
+    Add transition models
+    >>> intel_tm = {0: {0: 0.25, 1: 0.75}, 1: {0: 0.5, 1: 0.5}}
+    >>> model.add_transition_model('intel', intel_tm)
+    >>> diff_tm = {0: {0: 0.1, 1: 0.5, 2: 0.4}, 1: {0: 0.2, 1: 0.2, 2: 0.6 }, 2: {0: 0.7, 1: 0.15, 2: 0.15}}
+    >>> model.add_transition_model('diff', diff_tm)
+
+    Set a start state
+    >>> model.set_start_state({'intel': 0, 'diff': 2})
```

Comment:
@pratyakshs According to the new representation of states, is should be a tuple as (var_name, state).